sdk/core/azure-core-amqp/inc/azure/core/amqp/management.hpp

```diff
@@ -0,0 +1,157 @@
+       */
+      std::string ExpectedStatusDescriptionKeyName = "statusDescription";
+
+      /**
+       * @brief Enable trace logging for the management operations.
+       */
+      bool EnableTrace{false};
+    };
+
+    /**
+     * @brief Callback event handler for management events such as error.
+     */
+    struct ManagementEvents
+    {
+      /** @brief Called when an error occurs.
+       */
+      virtual void OnError() = 0;
+    };
+
+    /** @brief AMQP Management APIs.
+     *
+     * The AMQP management plane is a set of APIs that allow for management operations to be
+     * performed on an AMQP connection. See [AMQP Management
+     * Version 1.0](https://www.oasis-open.org/committees/download.php/52425/amqp-man-v1%200-wd08.pdf)
+     * for more information.
+     *
+     */
+    class Management final {
+    public:
+      /**
+       * @brief Create a new Management object instance.
+       *
+       * @param session - the session on which to create the instance.
+       * @param managementNodeName - the name of the message source and target.
+       * @param options - additional options for the Management object.
+       * @param managementEvents - events associated with the management object.
+       */
+      Management(
+          Session const& session,
+          std::string const& managementNodeName,
+          ManagementOptions const& options,
+          ManagementEvents* managementEvents = nullptr);
+
+      Management() = default;
+      Management(std::shared_ptr<Azure::Core::Amqp::_detail::ManagementImpl> impl) : m_impl{impl} {}
+      ~Management() noexcept = default;
+      operator bool() const;
+
+      /**
+       * @brief Open the management instance.
+       *
+       * @returns The result of the open operation.
+       */
+      ManagementOpenResult Open(Azure::Core::Context const& context = {});
+
+      /**
+       * @brief Close the management instance.
+       */
+      void Close();
+
+      /**
+       * @brief Execute a management operation.
+       *
+       * @param operationToPerform - the operation to perform (case sensitive).
+       * @param typeOfOperation - the type of operation (case sensitive).
+       * @param locales - the locales to use - A list of locales that the sending peer permits for
+       * incoming informational text in response messages. This value MUST be of the form presented
+       * in the "Language-Tag" rule of [RFC2616], section 3.10.
+       * @param messageToSend - the message to send.
+       * @param context - the context for the operation.
+       *
+       * @returns a tuple consisting of the high level result of the operation, the HTTP response
+       * status code, the status description, and the response message.
+       */
+      std::tuple<
+          _internal::ManagementOperationResult,
+          std::uint32_t,
+          std::string,
+          Models::AmqpMessage>
```

Comment:
With a tuple that is so large, maybe this could benefit from a strongly typed object, which encapsulate what executing a management operation could return.

I would have suggested `ManagementOperationResult`, but that's already a thing here.